juju/juju
# Make proxy settings truly global

## Description of change
Make proxy settings global - previously the settings were only used on 'ubuntu' user login shell, with this change the settings are global for all users using /etc/profile.d and, on systems using systemd, also set for services launched using systemd via /etc/systemd/{system,user}.conf.d/juju-proxy.conf. 

## QA steps
1. Run unit tests
2. Verify that if http-proxy option is set it's being set in the environment - e.g. services running via systemd should have proper env variables set.

## Documentation changes
This change has to be noted in release notes and docs.

## Bug reference
https://bugs.launchpad.net/juju/+bug/1666353
